crates/planus/src/impls/primitives.rs

```diff
@@ -4,9 +4,9 @@ use crate::{
     builder::Builder, errors::ErrorKind, slice_helpers::SliceWithStartOffset, traits::*, Cursor,
 };
 
-macro_rules! gen_primitive_types {
+macro_rules! unsafe_gen_primitive_types {
```

Comment:
In general, I need to document why these implementations work with safety comments both for the macros & the templates.

Reply:
Let me know if this is something you would like to get help/feedback on

Reply:
I will work on it, I think from Taylor's audit I have enough to explain why the current implementations are correct.